plugins/by-name/alpha/default.nix

```diff
@@ -193,19 +230,8 @@ in
         }
       ];
 
-      extraConfigLua =
-        let
-          setupOptions =
-            if themeDefined then
-              cfg.theme
-            else
-              (with cfg; {
-                inherit layout opts;
-              });
-        in
-        ''
-          require('alpha').setup(${lib.nixvim.toLuaObject setupOptions})
-          require('alpha.term')
-        '';
+      plugins.alpha.luaConfig.content =
+        optionalString themeDefined "require('alpha').setup(${toLuaObject cfg.theme})\n"
+        + "require('alpha.term')";
```

Comment:
I'm a little confused. You have a `settings` option, but you are also manually calling setup and passing it a `theme` option.

The idea of `settings` is that it is the nix representation of the lua passed to `setup`.

Reply:
`.setup` needs to be manually called when using `theme` option. Because that is the only way we can set pre-defined themes. That's why, `theme` and `settings.layout` options can't be used at the same time

Reply:
If so, that also implies:

1. `settings` as a whole must not be used when a theme is defined
2. `setup` must _always_ be called manually, either passing it `settings` or `theme`

Reply:
Interesting, the upstream examples show a different usage:

```lua
require('alpha').setup(require('alpha.themes.dashboard').config)
```

They are passing pre-defined settings to setup; getting those settings _from_ a theme module.

Reply:
It is the same usage, the theme option is being converted here:

https://github.com/nix-community/nixvim/blob/c83415c8c0125d22a7f79c3767d28775c696d5da/plugins/by-name/alpha/default.nix#L196-L201